sympy/printing/tests/test_maple.py

```diff
@@ -0,0 +1,413 @@
+
+
+def test_sparse():
+    M = SparseMatrix(5, 6, {})
+    M[2, 2] = 10
+    M[1, 2] = 20
+    M[1, 3] = 22
+    M[0, 3] = 30
+    M[3, 0] = x * y
+    assert maple_code(M) == \
+           'Matrix([[0, 0, 0, 30, 0, 0],' \
+           ' [0, 0, 20, 22, 0, 0],' \
+           ' [0, 0, 10, 0, 0, 0],' \
+           ' [x*y, 0, 0, 0, 0, 0],' \
+           ' [0, 0, 0, 0, 0, 0]], ' \
+           'storage = sparse)'
+
+
+def test_maple_not_supported():
+    assert maple_code(S.ComplexInfinity) == (
+        "# Not supported in maple:\n"
+        "# ComplexInfinity\n"
+        "zoo"
+    )
+
+
+
+def test_MatrixElement_printing():
+    # test cases for issue #11821
+    A = MatrixSymbol("A", 1, 3)
+    B = MatrixSymbol("B", 1, 3)
+    C = MatrixSymbol("C", 1, 3)
+
+    assert (maple_code(A[0, 0]) == "A[1, 1]")
+    assert (maple_code(3 * A[0, 0]) == "3*A[1, 1]")
+
+    F = C[0, 0].subs(C, A - B)
+
+    print(maple_code(F))
+    assert (maple_code(F) == "(A - B)[1, 1]")  # PROBLEM
+
+
+def test_hadamard():
+    A = MatrixSymbol('A', 3, 3)
+    B = MatrixSymbol('B', 3, 3)
+    v = MatrixSymbol('v', 3, 1)
+    h = MatrixSymbol('h', 1, 3)
+    C = HadamardProduct(A, B)
+    assert maple_code(C) == "A*B"
+
+    print(maple_code(C * v))
+    assert maple_code(C * v) == "(A*B)*v"  # PROBLEM
+    assert maple_code(h * C * v) == "h*(A*B)*v"
+    assert maple_code(C * A) == "(A*B)*A"
+    # mixing Hadamard and scalar strange b/c we vectorize scalars
+    assert maple_code(C * x * y) == "(x*y)*(A*B)"
+
+
+def test_maple_piecewise():
+    expr = Piecewise((x, x < 1), (x ** 2, True))
+
+    print(maple_code(expr))
+    assert maple_code(expr) == "piecewise(x < 1, x, x^2)"  # PROBLEM
+    assert maple_code(expr, assign_to="r") == (
+        "r := piecewise(x < 1, x, x^2)")
+
+    expr = Piecewise((x ** 2, x < 1), (x ** 3, x < 2), (x ** 4, x < 3), (x ** 5, True))
+    expected = "piecewise(x < 1, x^2, x < 2, x^3, x < 3, x^4, x^5)"
+    assert maple_code(expr) == expected
+    assert maple_code(expr, assign_to="r") == "r := " + expected
+
+    # Check that Piecewise without a True (default) condition error
+    expr = Piecewise((x, x < 1), (x ** 2, x > 1), (sin(x), x > 0))
+    raises(ValueError, lambda: maple_code(expr))
+
+
+def test_maple_piecewise_times_const():
+    pw = Piecewise((x, x < 1), (x ** 2, True))
+
+    print(maple_code(2 * pw))
+    assert maple_code(2 * pw) == "2*piecewise(x < 1, x, x^2)"  # PROBLEM
+    assert maple_code(pw / x) == "piecewise(x < 1, x, x^2)/x"
+    assert maple_code(pw / (x * y)) == "piecewise(x < 1, x, x^2)/(x*y)"
+    assert maple_code(pw / 3) == "piecewise(x < 1, x, x^2)/3"
+
+
+def test_maple_derivatives():
+    f = Function('f')
+
+    print(maple_code(f(x).diff(x)))
+    print(maple_code(f(x).diff(x, 2)))
+    assert maple_code(f(x).diff(x)) == 'diff(f(x), x)'
+    assert maple_code(f(x).diff(x, 2)) == 'diff(f(x), x$2)'
+
+
+def test_specfun():
+    # FIXME: don't know if maple support these feature.
```

Comment:
https://www.maplesoft.com/support/help/Maple/view.aspx?path=Bessel
https://www.maplesoft.com/support/help/Maple/view.aspx?path=Airy

It may work to just add them to the `known_functions` translation table. But feel free to skip them for now. Comparing with the Mathematica printer, it was just quite recently that I added support for those, so that had existed for a long time without them.

Reply:
OK, understood.

Reply:
I will add them.